exporter/exporterhelper/internal/queue_sender.go

```diff
@@ -127,18 +130,41 @@ func (qs *QueueSender) Start(ctx context.Context, host component.Host) error {
 	}
 
 	dataTypeAttr := attribute.String(DataTypeKey, qs.obsrep.Signal.String())
-	return multierr.Append(
-		qs.obsrep.TelemetryBuilder.InitExporterQueueSize(func() int64 { return int64(qs.queue.Size()) },
-			metric.WithAttributeSet(attribute.NewSet(qs.traceAttribute, dataTypeAttr))),
-		qs.obsrep.TelemetryBuilder.InitExporterQueueCapacity(func() int64 { return int64(qs.queue.Capacity()) },
-			metric.WithAttributeSet(attribute.NewSet(qs.traceAttribute))),
-	)
+
+	reg1, err1 := qs.obsrep.TelemetryBuilder.InitExporterQueueSize(func() int64 { return int64(qs.queue.Size()) },
+		metric.WithAttributeSet(attribute.NewSet(qs.traceAttribute, dataTypeAttr)))
+
+	if reg1 != nil {
+		qs.shutdownFns = append(qs.shutdownFns, func(context.Context) error {
+			return reg1.Unregister()
+		})
+	}
+
+	reg2, err2 := qs.obsrep.TelemetryBuilder.InitExporterQueueCapacity(func() int64 { return int64(qs.queue.Capacity()) },
+		metric.WithAttributeSet(attribute.NewSet(qs.traceAttribute)))
+
+	if reg2 != nil {
+		qs.shutdownFns = append(qs.shutdownFns, func(context.Context) error {
+			return reg2.Unregister()
+		})
+	}
+
+	return multierr.Append(err1, err2)
```

Comment:
You can remove the multierr external package. The Go errors lib now includes this natively.
https://pkg.go.dev/errors#Join

```suggestion
	return errors.Join(err1, err2)
```

Reply:
done